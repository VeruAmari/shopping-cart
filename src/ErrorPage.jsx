import { Link } from 'react-router-dom';
import { Heading } from './components/sub/Headers';
import { PageContainer } from './components/main/App';

const ErrorPage = () => {
  return (
    <PageContainer>
      <Heading level="1">Oh no! This route doesn't exist!</Heading>
      <Link to="/">You can go back to the home page by clicking here.</Link>
    </PageContainer>
  );
};

export default ErrorPage;
