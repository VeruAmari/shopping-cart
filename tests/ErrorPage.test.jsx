import { render, screen } from '@testing-library/react';
import ErrorPage from '../src/ErrorPage';

import { MemoryRouter } from 'react-router-dom';
describe('ErrorPage', () => {
  it('renders', () => {
    render(
      <MemoryRouter>
        <ErrorPage>Error</ErrorPage>
      </MemoryRouter>,
    );
    expect(screen.getByText('Error')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    render(
      <MemoryRouter>
        <ErrorPage>Error</ErrorPage>
      </MemoryRouter>,
    );
    expect(screen.getByText('Error')).toMatchSnapshot();
  });
});
