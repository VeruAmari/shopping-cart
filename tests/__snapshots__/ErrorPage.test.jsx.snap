// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > matches snapshot 1`] = `
<div
  class="App__PageContainer-sc-9o5ywn-1 hRIkIP"
>
  <h1
    class="Headers__Wrapper-sc-o55uf2-0 ceolFo"
    style="margin: 0.5rem;"
  >
    Oh no! This route doesn't exist!
  </h1>
  Error
  <a
    href="/"
  >
    You can go back to the home page by clicking here.
  </a>
</div>
`;
